In [2]:
from pathlib import Path

import pandas as pd

NOTEBOOK_DIR = Path.cwd()
PROJECT_ROOT = NOTEBOOK_DIR.parent if NOTEBOOK_DIR.name == "notebooks" else NOTEBOOK_DIR
PARQUET_PATH = PROJECT_ROOT / "data" / "raw" / "openmeteo_finale.parquet"

df = pd.read_parquet(PARQUET_PATH)

In [3]:
df.shape

(3495600, 38)

**Chargement et vue d'ensemble
Vérification de la taille et structure du dataset avant tout nettoyage.**

In [8]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 3495600 entries, 0 to 3495599
Data columns (total 38 columns):
 #   Column                          Dtype  
---  ------                          -----  
 0   time                            str    
 1   weather_code                    float64
 2   temperature_2m_mean             float64
 3   temperature_2m_min              float64
 4   temperature_2m_max              float64
 5   apparent_temperature_mean       float64
 6   apparent_temperature_min        float64
 7   apparent_temperature_max        float64
 8   relative_humidity_2m_mean       float64
 9   relative_humidity_2m_min        float64
 10  relative_humidity_2m_max        float64
 11  dew_point_2m_mean               float64
 12  precipitation_sum               float64
 13  rain_sum                        float64
 14  snowfall_sum                    float64
 15  precipitation_hours             float64
 16  wind_speed_10m_mean             float64
 17  wind_speed_10m_max              float6

**Pour vérifier les colonnes, les types et les valeurs manquantes.**

In [4]:
df.head()

,time,weather_code,temperature_2m_mean,temperature_2m_min,temperature_2m_max,apparent_temperature_mean,apparent_temperature_min,apparent_temperature_max,relative_humidity_2m_mean,relative_humidity_2m_min,...,soil_moisture_28_to_100cm_mean,soil_temperature_0_to_7cm_mean,nom_poi,numero_departement,latitude_poi,longitude_poi,Ville,Departement,Latitude,Longitude
0,2000-01-01,51.0,7.3,5.8,9.0,5.5,3.8,7.4,96.0,92.0,...,0.346,6.4,Melun,77,48.541637,2.658873,NaN,NaN,NaN,NaN
1,2000-01-02,3.0,7.5,6.7,9.1,5.8,4.6,7.5,96.0,88.0,...,0.344,7.1,Melun,77,48.541637,2.658873,NaN,NaN,NaN,NaN
2,2000-01-03,3.0,6.5,5.5,8.1,3.3,2.0,5.3,88.0,82.0,...,0.342,6.5,Melun,77,48.541637,2.658873,NaN,NaN,NaN,NaN
3,2000-01-04,53.0,7.6,5.9,9.3,3.7,1.6,6.6,89.0,83.0,...,0.340,6.8,Melun,77,48.541637,2.658873,NaN,NaN,NaN,NaN
4,2000-01-05,3.0,7.7,7.1,8.6,5.1,4.5,5.9,91.0,83.0,...,0.338,7.3,Melun,77,48.541637,2.658873,NaN,NaN,NaN,NaN


In [9]:
df["lieu"] = df["nom_poi"].fillna(df["Ville"])

**Je réunis deux colonnes qui contiennent les villes.**

In [10]:
df["lieu"].nunique()

360

**On compte le nombre de lieux distincts après fusion de `nom_poi` et `Ville`.**

On obtient 360 lieux uniques, ce qui correspond au nombre de POI et de villes du jeu de données source.

In [11]:
df["time"].min(), df["time"].max()

('2000-01-01', '2026-08-01')

**On vérifie la période couverte par les données.**

Les dates s'étendent du 2000-01-01 au 2026-08-01.

In [12]:
compte = df.groupby("lieu")["time"].nunique()

In [13]:
compte[compte != 9710]

Series([], Name: time, dtype: int64)

**Les 360 lieux possèdent tous exactement 9710 dates uniques. Aucun lieu n'a de date manquante sur la période.**

In [14]:
compte.describe()

count     360.0
mean     9710.0
std         0.0
min      9710.0
25%      9710.0
50%      9710.0
75%      9710.0
max      9710.0
Name: time, dtype: float64

In [15]:
df.duplicated(subset=["lieu", "time"]).sum()

np.int64(0)

**On contrôle qu'il n'existe pas de doublon lieu + date.**

Aucun doublon détecté.

In [16]:
df.isna().sum().sort_values(ascending=False)

Latitude                          2592570
Longitude                         2592570
Ville                             2592570
Departement                       2592570
numero_departement                 903030
latitude_poi                       903030
longitude_poi                      903030
nom_poi                            903030
soil_temperature_0_to_7cm_mean        120
temperature_2m_max                    120
apparent_temperature_min              120
temperature_2m_min                    120
temperature_2m_mean                   120
weather_code                          120
apparent_temperature_mean             120
apparent_temperature_max              120
relative_humidity_2m_mean             120
relative_humidity_2m_min              120
relative_humidity_2m_max              120
dew_point_2m_mean                     120
precipitation_sum                     120
rain_sum                              120
snowfall_sum                          120
et0_fao_evapotranspiration        

In [17]:
sorted(df["numero_departement"].dropna().unique())

['01',
 '02',
 '03',
 '04',
 '05',
 '06',
 '07',
 '08',
 '09',
 '10',
 '11',
 '12',
 '13',
 '14',
 '15',
 '16',
 '17',
 '18',
 '19',
 '21',
 '22',
 '23',
 '24',
 '25',
 '26',
 '27',
 '28',
 '29',
 '2A',
 '2B',
 '30',
 '31',
 '32',
 '33',
 '34',
 '35',
 '36',
 '37',
 '38',
 '39',
 '40',
 '41',
 '42',
 '43',
 '44',
 '45',
 '46',
 '47',
 '48',
 '49',
 '50',
 '51',
 '52',
 '53',
 '54',
 '58',
 '59',
 '60',
 '67',
 '68',
 '69',
 '70',
 '74',
 '77',
 '78',
 '79',
 '80',
 '81',
 '82',
 '83',
 '84',
 '85',
 '86',
 '87',
 '88',
 '89',
 '90',
 '91',
 '92',
 '93',
 '94',
 '95']

**On liste les codes département présents dans le jeu de données.**

82 départements distincts sont couverts, avec des codes qui semblent cohérents (pas de valeur aberrante).

In [18]:
# Lignes où les données météo sont manquantes
nan_meteo = df[df["temperature_2m_mean"].isna()]

In [19]:
len(nan_meteo)

120

**On chiffre précisément le volume de lignes météo manquantes.**

120 lignes présentent des valeurs météo manquantes, toutes sur `temperature_2m_mean` et les colonnes météo associées.

In [20]:
nan_meteo[["time", "lieu"]]

,time,lieu
9709,2026-08-01,Melun
19419,2026-08-01,Torcy
29129,2026-08-01,Rambouillet
38839,2026-08-01,Parthenay
48549,2026-08-01,Niort
...,...,...
1126359,2026-08-01,Troyes
1136069,2026-08-01,Bar-sur-aube
1145779,2026-08-01,Limoux
1155489,2026-08-01,Narbonne


In [ ]:
nan_meteo["time"].value_counts()

**Les valeurs manquantes sont toutes le même jour, le 01 aout 2026.**

In [21]:
nan_meteo["lieu"].nunique()

120

**Bilan de l'investigation des NaN météo.**

Les 120 lignes manquantes correspondent à 120 lieux distincts (aucun lieu n'apparaît deux fois dans ce sous-ensemble), et concernent toutes la même date, le 2026-08-01 — le tout dernier jour disponible dans le jeu de données. Cela ressemble à une journée partiellement collectée plutôt qu'à un problème réparti dans le temps. Ces lignes ne sont ni supprimées ni corrigées ici : leur traitement relève du staging dbt.